luigi/scheduler.py

```diff
@@ -655,9 +672,15 @@ def add_task(self, task_id=None, status=PENDING, runnable=True,
         if deps is not None:
             task.deps = set(deps)
 
+        if deps_configs is not None:
+            task.deps_configs = deps_configs
+
         if new_deps is not None:
             task.deps.update(new_deps)
 
+        if new_deps_configs is not None:
+            task.deps_configs.update(new_deps_configs)
```

Comment:
I suppose that they do have task-level config too. Makes sense right? However, from this code (when you read this in combination with the `worker.py` code related to the `new_deps_configs` variable) it seems like you "inherit" the task-level config which just is nonsensical, right? I might have misunderstood your code though. :)